specification/applicationinsights/ApplicationInsights.Monitor.Exporters/models.tsp

```diff
@@ -0,0 +1,817 @@
+
+  /** User account ID. */
+  AiUserAccountId: "ai.user.accountId",
+
+  /** User ID. */
+  AiUserId: "ai.user.id",
+
+  /** Authenticated user ID. */
+  AiUserAuthUserId: "ai.user.authUserId",
+
+  /** Cloud role. */
+  AiCloudRole: "ai.cloud.role",
+
+  /** Cloud role version. */
+  AiCloudRoleVer: "ai.cloud.roleVer",
+
+  /** Cloud role instance. */
+  AiCloudRoleInstance: "ai.cloud.roleInstance",
+
+  /** Cloud location. */
+  AiCloudLocation: "ai.cloud.location",
+
+  /** Internal SDK version. */
+  AiInternalSdkVersion: "ai.internal.sdkVersion",
+
+  /** Internal agent version. */
+  AiInternalAgentVersion: "ai.internal.agentVersion",
+
+  /** Internal node name. */
+  AiInternalNodeName: "ai.internal.nodeName",
+}
+
+/**
+ * System variables for a telemetry item.
+ */
+model TelemetryEnvelope {
+  /**
+   * Envelope version. For internal use only. By assigning this the default, it will
+   * not be serialized within the payload unless changed to a value other than #1.
+   */
+  ver?: int32 = 1;
+
+  /**
+   * Type name of telemetry data item.
+   */
+  name: string;
+
+  /**
+   * Event date time when telemetry item was created. This is the wall clock time on
+   * the client when the event was generated. There is no guarantee that the
+   * client's time is accurate. This field must be formatted in UTC ISO 8601 format,
+   * with a trailing 'Z' character, as described publicly on
+   * https://en.wikipedia.org/wiki/ISO_8601#UTC. Note: the number of decimal seconds
+   * digits provided are variable (and unspecified). Consumers should handle this,
+   * i.e. managed code consumers should not use format 'O' for parsing as it
+   * specifies a fixed length. Example: 2009-06-15T13:45:30.0000000Z.
+   */
+  time: utcDateTime;
+
+  /**
+   * Sampling rate used in application. This telemetry item represents 100 /
+   * sampleRate actual telemetry items.
+   */
+  sampleRate?: float32 = 100;
+
+  /**
+   * Sequence field used to track absolute order of uploaded events.
+   */
+  @maxLength(64)
+  seq?: string;
+
+  /**
+   * The instrumentation key of the Application Insights resource.
+   */
+  iKey?: string;
+
+  /**
+   * Key/value collection of context properties. See ContextTagKeys for information
+   * on available properties.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Telemetry data item.
+   */
+  data?: Base;
+}
+
+/**
+ * Data struct to contain only C section with custom fields.
+ */
+model Base {
+  /**
+   * Name of item (B section) if any. If telemetry data is derived straight from
+   * this, this should be null.
+   */
+  baseType?: string;
+
+  /**
+   * The data payload for the telemetry request
+   */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For backward compatibility"
+  baseData?: Domain;
+}
+
+/**
+ * The abstract common base of all domains.
+ */
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For backward compatibility"
+model Domain {
+  ...Record<unknown>;
```

Comment:
@catalinaperalta, is this the way to define that the model can have arbitrary fields in it?

Reply:
This will mark that the model has additional properties, so if that's the goal then yes it is.